backend/src/main/java/gov/cdc/usds/simplereport/service/DataHubUploaderService.java

```diff
@@ -0,0 +1,137 @@
+package gov.cdc.usds.simplereport.service;
+
+import com.fasterxml.jackson.dataformat.csv.CsvGenerator;
+import com.fasterxml.jackson.dataformat.csv.CsvMapper;
+import com.fasterxml.jackson.dataformat.csv.CsvSchema;
+import gov.cdc.usds.simplereport.api.model.TestEventExport;
+import gov.cdc.usds.simplereport.db.model.TestEvent;
+import gov.cdc.usds.simplereport.db.model.auxiliary.RaceArrayConverter;
+import gov.cdc.usds.simplereport.db.repository.TestEventRepository;
+import net.minidev.json.JSONObject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.web.client.RestTemplateBuilder;
+import org.springframework.core.io.ByteArrayResource;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.MediaType;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.web.client.RestClientException;
+import org.springframework.web.client.RestTemplate;
+
+import javax.persistence.NoResultException;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.text.SimpleDateFormat;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+import java.util.List;
+
+@Service
+@Transactional
+public class DataHubUploaderService {
+    @Value("${simple-report.data-hub.uploadurl}")
+    private String DATAHUB_UPLOAD_URL;
+    static final int MAX_ROWS_ALLOWED_PER_BATCH = 999;
+    static final String API_VERSION = "2";
+    private static final Logger LOG = LoggerFactory.getLogger(RaceArrayConverter.class);
+
+    private final TestEventRepository _repo;
+    private String _fileContents;
+    private String _nextTimestamp;
+    private String _warnMessage;
+    private String _uploadResultId;
+    private int _rowCount;
+
+    public DataHubUploaderService(TestEventRepository _repo) {
+        this._repo = _repo;
+        this._nextTimestamp = "";
+        this._warnMessage = "";
+        this._uploadResultId = "";
+        this._rowCount = 0;
+    }
+
+    private void _createTestEventCSV(String lastEndCreateOn) throws IOException, DateTimeParseException, NoResultException {
+        if (lastEndCreateOn.length() == 0) {
+            // testing only, just query everything
+            lastEndCreateOn = "2020-01-01T00:00:00.00Z";
+            LOG.info("Empty startupdateby so using 2020-01-01");
+        }
+        List<TestEvent> events = _repo.findAllByCreatedAtInstant(lastEndCreateOn);
+        if (events.size() == 0) {
+            throw new NoResultException();
```

Comment:
Why would this be exceptional?

Reply:
Heh. You're right, it probably doesn't need to be. I'll fix it.

Reply:
OK. Looking back over the code, throwing here improves readability in the higher level's error handling code. I'd like to leave it for now and revisit the correct way to deal with empty results.